pkg/hive/hivetest/health.go

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package hivetest
+
+import "github.com/cilium/cilium/pkg/lock"
+
+type MockHealthReporter struct {
+	lock.Mutex
+	m       string
+	stopped bool
+	err     error
+}
+
+func (m *MockHealthReporter) Msg() string {
+	m.Lock()
+	defer m.Unlock()
+	return m.m
+}
+
+func (m *MockHealthReporter) Err() error {
+	m.Lock()
+	defer m.Unlock()
+	return m.err
+}
+
+func (m *MockHealthReporter) IsStopped() bool {
```

Comment:
This is personal preference, but IMO this reads a little more naturally and makes for a shorter method name:
```
if m.Stopped() {
  do something
}
```

```suggestion
func (m *MockHealthReporter) Stopped() bool {
```